### DISCUSSION TITLE
Alternate Function Invocation Syntax for SQL Server

### DISCUSSION THREAD

**Discussion by jdejesus517** [sqlalchemy/sqlalchemy#6745] [Usage Questions] [answered]
I'm trying to write custom functions and custom types to leverage SQL Server spatial functions. Switching to PostGIS is not an option.

SQL Server syntax for a spatial function operates on the column rather than having a function where columns are passed in.
For example:
```
SELECT geography_column.STBuffer(3) FROM some_table;
```
or
```
SELECT geom1.STIntersects(geom2) FROM some_table;
```

Using `func` just passes the columns or other values to the function you normally would expect but I need something that creates the dot syntax style of invocation. I've been looking through the source on the function creation classes and I can't make heads or tails of what I may need to subclass/modify to augment the behavior to get the right output.

Any thoughts from the community?

**Comment by zzzeek** [comment, marked as answer]
we have an issue for this in #5006 and there is no plan right now to work on it however we are open to contributions.  ideally we'd like geoalchemy to imlpement something for this but in the meantime you basically make yourself a custom SQL construct as illustrated in that issue.
